unattended_scripts/install_functions/opendistro/common.sh

```diff
@@ -46,48 +46,66 @@ checkSystem() {
 
 checkNames() {
 
-    if [[ "${einame}" == "${kiname}" ]] || [[ "${einame}" == "${winame}" ]] || [[ "${kiname}" == "${winame}" ]]; then
+    if [[ -n ${einame} ]] && [[ -n ${kiname} ]] && ([[ "${einame}" == "${kiname}" ]]); then
         logger -e "The node names for Elastisearch, Kibana and Wazuh must be different."
         exit 1
     fi
 
-    if [ -n ${einame} ]; then
+    if [[ -n ${einame} ]] && [[ -n ${winame} ]] && ([[ "${einame}" == "${winame}" ]]); then
+        logger -e "The node names for Elastisearch, Kibana and Wazuh must be different."
+        exit 1
+    fi
+
+    if [[ -n ${winame} ]] && [[ -n ${kiname} ]] && ([[ "${winame}" == "${kiname}" ]]); then
+        logger -e "The node names for Elastisearch, Kibana and Wazuh must be different."
+        exit 1
+    fi
+
+    if [[ -n ${einame} ]]; then
         if [[ ! "${elasticsearch_node_names[@]}" =~ "${einame}" ]]; then
-            logger -e "The name given for the elasticsearch node does not appear on the configuration file."
+            logger -e "The name given for the ElasticSearch node does not appear on the configuration file."
             exit 1
         fi
+    fi
 
-        if [ ! -f ${base_path}/certs/${einame}.pem ] || [ ! -f ${base_path}/certs/${einame}-key.pem ]; then
-            logger -e "There is no certificate for the elasticsearch node ${einame} in ${base_path}/certs."
+    if [[ -n ${winame} ]]; then
+        if [[ ! "${wazuh_servers_node_names[@]}" =~ "${winame}" ]]; then
+            logger -e "The name given for the Wazuh server node does not appear on the configuration file."
             exit 1
         fi
-
     fi
 
-    if [ -n ${winame} ]; then
-        if [[ ! "${wazuh_servers_node_names[@]}" =~ "${winame}" ]]; then
-            logger -e "The name given for the wazuh server node does not appear on the configuration file."
+    if [[ -n ${kiname} ]]; then
+        if [[ ! "${kibana_node_names[@]}" =~ "${kiname}" ]]; then
+            logger -e "The name given for the Kibana node does not appear on the configuration file."
             exit 1
         fi
+    fi
 
-        if [ ! -f ${base_path}/certs/${winame}.pem ] || [ ! -f ${base_path}/certs/${winame}-key.pem ]; then
-            logger -e "There is no certificate for the wazuh server node ${winame} in ${base_path}/certs."
+}
+
+checkPreviousCertificates() {
+
+    if [ -n ${einame} ]; then
+        if [ -f ${base_path}/certs/${einame}.pem ] || [ -f ${base_path}/certs/${einame}-key.pem ]; then
+            logger -e "There is a certificate for the elasticsearch node ${einame} in ${base_path}/certs."
```

Comment:
If you capitalize Kibana, then all references to Wazuh, Kibana, Elasticsearch and Filebeat must be capitalized

Reply:
Done